File: biddo_client/lib/core/repositories/comment.dart
import 'package:get/get.dart';

import '../models/comment.dart';
import 'base.dart';

class CommentRepository {
  var dio = Get.find<Api>();

  Future<Comment?> create(
    String content,
    String auctionId, [
    String? parentCommentId,
  ]) async {
    var commentContent = content.replaceAll('\n\n\n\n\n\n', '\n');

    try {
      var result = await dio.api.post('/comment', data: {
        "auctionId": auctionId,
        "content": commentContent,
        "parentCommentId": parentCommentId
      });

      return Comment.fromJSON(result.data);
    } catch (error) {
      print('Could not create comment: $error');
      return null;
    }
  }

  Future<List<Comment>?> getComments(String auctionId) async {
    try {
      var result = await dio.api.get('/comment/auction/$auctionId');

      return List<Comment>.from(
          result.data.map((comment) => Comment.fromJSON(comment)));
    } catch (error) {
      print('Could not get comments for auction: $error');
      return null;
    }
  }

  Future<int> countForAuction(String auctionId) async {
    try {
      var result = await dio.api.get('/comment/count/$auctionId');
      return result.data;
    } catch (error) {
      print('Could not get comment count for auction: $error');
      return 0;
    }
  }

  Future<String?> translate(String commentId, String lang) async {
    try {
      var response = await dio.api.get('/comment/translate/$commentId/$lang');
      return response.data['content'];
    } catch (error, stackTrace) {
      print('Error translating comment: $error $stackTrace');
      return null;
    }
  }
}

File: core/repositories/comment.ts
import { RequestMaker, RequestType } from '../services/request-maker'
import { Comment } from '../domain/comment'

class CommentRepositroy {
  private basePath = '/comment'

  public async create(content: string, auctionId: string, parentCommentId?: string) {
    try {
      const result = await RequestMaker.makeRequest({
        path: `${this.basePath}`,
        method: RequestType.POST,
        payload: JSON.stringify({
          content,
          auctionId,
          parentCommentId,
        }),
      })
      return Comment.fromJSON(result as Record<string, unknown>)
    } catch (error) {
      console.error('Error creating comment:', error)
      return null
    }
  }

  public async getComments(auctionId: string) {
    try {
      const response = await RequestMaker.makeRequest({
        path: `${this.basePath}/auction/${auctionId}`,
        method: RequestType.GET,
      })
      return (response as Record<string, unknown>[]).map((el) => Comment.fromJSON(el))
    } catch (error) {
      console.error('Error getting comments:', error)
      return []
    }
  }

  public async countForAuction(auctionId: string) {
    try {
      const response = await RequestMaker.makeRequest({
        path: `${this.basePath}/count/${auctionId}`,
        method: RequestType.GET,
      })
      return response as number
    } catch (error) {
      console.error('Error counting comments:', error)
      return 0
    }
  }

  public async translate(commentId: string, lang: string) {
    try {
      const response = (await RequestMaker.makeRequest({
        path: `${this.basePath}/translate/${commentId}/${lang}`,
        method: RequestType.GET,
      })) as { content: string }

      return response.content as string
    } catch (error) {
      console.error('Error translating comment:', error)
      return null
    }
  }
}

const repository = new CommentRepositroy()
export { repository as CommentRepositroy }
